digital-deck-server/services/player.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Class for player data and communications.
+ */
+module.exports = class Player {
+  // keep track of distinct players
+  static players = 0;
+  constructor(cards) {
+    this.cards = cards;
+    this.playerId = Player.players++;
+  }
+
+  /**
+   * Resets the ids.
+   * */
+  static resetPlayerCount() {
+    Player.players = 0;
+  }
+
+  /**
+   * Replace current cards with a new Array of cards
+   *
+   * @param cards The new cards array to replace the older one.
+   */
+  updateCards(cards) {
+    this.cards = cards;
+  }
+
+  /**
+   * add a card to bottom of this players stack
+   *
+   * @param card the card to be added
+   */
+  addCardBottom(card) {
+    this.cards.splice(0, 0, card);
+  }
+
+  /**
+   * add a card to top of this players stack
+   *
+   * @param card the card to be added
+   */
+  addCardTop(card) {
+    this.cards.push(card);
+  }
+
+  /**
+   * add a card to desired position of this players stack
+   *
+   * @param card the card to be added
+   */
+  addCard(card, index) {
```

Comment:
I don't think this is adding a card at the desired index. The param is not being used.